src/main/resources/feedbackQuestionRankOptionsEditFormTemplate.html

```diff
@@ -29,7 +29,7 @@
       <div class="col-sm-9 col-xs-12">
         <div class="checkbox" data-toggle="tooltip" data-placement="top" data-container="body" title="Ticking this will ensure respondent ranks at least the mentioned number of options.">
           <label class="bold-label">
-            <input type="checkbox" name="${Const.ParamsNames.FEEDBACK_QUESTION_RANKMINOPTIONSCHECKBOX}"
+            <input class="ranked-question-options-checkbox" type="checkbox" name="${Const.ParamsNames.FEEDBACK_QUESTION_RANKMINOPTIONSCHECKBOX}"
```

Comment:
Why is there a need to introduce a new class when there is already an id?

Reply:
That's a very good point :smiley: will need to change that to reduce unnecessary dependency on class

Reply:
The required interaction is that for all ranked question checkbox ranked-recipients or ranked-options, it automatically sets value of corresponding input to 1.

Currently the id is set to be `id="${Const.ParamsNames.FEEDBACK_QUESTION_RANKMINOPTIONSCHECKBOX}-${questionNumber}"`.

On the other hand, I need to specify this interaction regardless of questionNumber. More specifically, I need to set up this code during page setup:

![image](https://user-images.githubusercontent.com/4986717/36295411-660c8f02-1322-11e8-977d-3975a90dc840.png)

Is it appropriate to use class in this case then?

Reply:
If not using something like a jquery starts with selector is also possible. But semantically, i am not sure whether using class for something more general (applicable to all corresponding checkboxes) is appropriate.

Reply:
Correct me if I'm wrong, but don't you want to do this only for the currently edited question only?

Would using `id` not work?

Reply:
Using id would work, just need to change the code to use jquery startsWith attribute selector for id.

The intended outcome is to add the interaction shown in the code for all checkbox that has that class or id starting with `{Const.ParamsNames.FEEDBACK_QUESTION_RANKMINOPTIONSCHECKBOX}`. Specifically, clicking on such a checkbox would set value to 1 if it is empty. 

The discussion is whether using class is more appropriate compared to id semantically because this needs to be added to all such checkboxes and not for a specific question. For example, there are 3 ranked questions in a previous feedback session. Whenever the instructor is editing any of the previous questions as well as new question, this same interaction would work.

If class is reserved for styling purposes, then I believe using id would be more appropriate in this case.

Reply:
`class` is not reserved for only styling purposes, but let's not go there yet :P 

How about using id by passing along the currently edited question/when creating a new question?

Somewhere here?
```js
function enableEdit(questionNum, maxQuestions) {
    let i = maxQuestions;
    while (i) {
        if (questionNum === i) {
            backupQuestion(i);
            enableQuestion(i);
        } else {
            disableQuestion(i);
        }
        i -= 1;
    }

    return false;
}
```

Reply:
Hmm okay. I think it's possible to not pass along currently "focused" question. I will work out a solution that uses `id` only.

Reply:
You mean it's possible, right?

Reply:
Yes I mean I think it's possible to achieve using `id` but not by passing along the question.

Reply:
If it's not possible it may be fine to just use `class`? But why is that so?

Reply:
Sorry for being unclear haha. I mean I think it's possible, I will make the change soon. However, I don't think we have to go so far as to pass along the currently edited question to make the change.

Reply:
Anyhow, this is what jQuery says:

> This selector can be useful for identifying elements in pages produced by server-side frameworks that produce HTML with systematic element IDs. However it will be slower than using a class selector so leverage classes, if you can, to group like elements.

https://api.jquery.com/attribute-starts-with-selector/

Reply:
Yes I think we definitely need to be careful when using startsWith. On the other hand ${Const.ParamsNames.FEEDBACK_QUESTION_RANKMINOPTIONSCHECKBOX} seems pretty precise as it really mean checkbox, but we need to consider future changes as well.

As for performance, we need to check whether matters (whether it can be felt by user). Passing question along for this specific use case can make the codebase more complicated.

Reply:
> ${Const.ParamsNames.FEEDBACK_QUESTION_RANKMINOPTIONSCHECKBOX}

It is precise, but we currently have a lot of trouble if we want to change it because we are passing magic strings and numbers everywhere. If we do fuzzy matching again, we are banking on the fact that future changes cannot be done anyway due to the previous design.

> As for performance, we need to check whether matters (whether it can be felt by user).

Let's not discuss this but performance in Javascript is probably more troublesome than most applications (We are still supporting mobile aren't we?).

> Passing question along for this specific use case can make the codebase more complicated.

I don't feel this is the case though this is totally up to you :).